File: session-03-who-owns-the-main-loop/typescript/src/order/order.service.ts
import {Injectable, OnModuleDestroy, OnModuleInit} from "@nestjs/common";
import {Order} from "./order.model";
import {OrderRepository} from "./order.repository";

// OnModuleInit and OnModuleDestroy are NestJS lifecycle hooks.
// The framework calls these at startup and shutdown.
@Injectable()
export class OrderService implements OnModuleInit, OnModuleDestroy {
    constructor(private readonly repository: OrderRepository) {
    }

    onModuleInit() {
        console.log("[Lifecycle] OrderService initialized");
    }

    onModuleDestroy() {
        console.log("[Lifecycle] OrderService shutting down");
    }

    listOrders(): Order[] {
        return this.repository.findAll();
    }

    findOrder(id: number): Order | undefined {
        return this.repository.findById(id);
    }
}

File: session-03-who-owns-the-main-loop/java/src/main/java/coffeeshop/order/OrderService.java
package coffeeshop.order;

import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class OrderService {

    private final OrderRepository repository;

    public OrderService(OrderRepository repository) {
        this.repository = repository;
    }

    public List<Order> listOrders() {
        return repository.findAll();
    }

    public Optional<Order> findOrder(int id) {
        return repository.findById(id);
    }
}
